httpcore/_sync/http2.py

```diff
@@ -572,6 +575,12 @@ def __iter__(self) -> typing.Iterator[bytes]:
             # we want to close the response (and possibly the connection)
             # before raising that exception.
             with ShieldCancellation():
+                # need send cancel frame when the exception is not from remote peer.
+                if not isinstance(exc, RemoteProtocolError):
```

Comment:
> need send cancel frame when the exception is not from remote peer.

It's not obvious to me that `not isinstance(exc, RemoteProtocolError)` is correct/sufficient here.

Should this instead be something like *just wrapping the yield*?...

```python
try:
    yield chunk
except Exception as exc:
    self._connection._reset_steam(
        stream_id=self._stream_id,
        error_code=h2.settings.ErrorCodes.CANCEL,
    )
    raise exc
```